### DISCUSSION TITLE
URL playback latency / buffer size

### DISCUSSION THREAD

**Discussion by matejdro** [music-assistant/support#2253] [Q&A] [answered]
I'm considering migrating from MDP to Music Assistant and I'm wondering if there is a way to control buffer size when playing from URL (http radio stations)? I'm interested in minimizing buffer size as much as possible to lower the latency, as they are usually quite high (on the order of seconds).

**Comment by OzGav** [comment]
No but MA will be optimised on a per player type basis so you should not have to do this.

**Reply by matejdro** [reply]
My use case for this is streaming from the phone: https://github.com/orgs/music-assistant/discussions/1757#discussioncomment-9415524, where latency is pretty big annoyance, but buffering is not needed as much, because it's only local network.

What do you mean by "optimised on a per player type basis"?

**Reply by OzGav** [reply]
Each player type will have buffers set depending on what it can accept as I understand it. 

You cannot steam from your phone at the moment to MA unless your phone generates a streaming URL as then you could add it as a radio station but I doubt it does that.

**Reply by matejdro** [reply]
Yes, check the linked discussion, I'm generating URL via VLC. With VLC I can set the buffer size, but the receiving MPD (that I'm currently using) adds extra buffers, which incur latency. I was hoping Music Assistant would be better in this regard, which would make it an instant switch for me.

**Reply by OzGav** [reply]
I think this might be an XY problem. Please explain the outcome you want rather than asking how to make what you currently do work.

**Reply by matejdro** [reply]
It is an XY problem :)

X is described here: https://github.com/orgs/music-assistant/discussions/1757

But since X is not available, I was hoping I could achieve this via Y (URL playback). I'm not aware of any other way to get a stream of audio into MA.

**Reply by OzGav** [reply]
OK so I understand you have something that you want to stream to a player and you are using VLC to do that. Why not just have MA stream whatever this is directly?

**Reply by matejdro** [reply]
It's explained in the linked discussion. There are two use cases here:

* Podcasts. As far as I can tell, there is no good self hosted podcast manager at all, let alone one that would integrate with MA. So the next best thing is to use one of the myriad podcasting apps on my phone and then stream the podcasts from my phone to the MA if I want to listen them through my sound system.
* Guests. If a guest wants to play some of his/her music, it is far easier to just let him/her stream it from her phone rather than explaining the MA interface. Plus there is a chance that MA might not have the music that the guest wants, while the guest likely has it on his/her phone.

**Reply by OzGav** [reply, marked as answer]
Both of those things are on the roadmap for the future. In regards to the original post that is not something beling considered at this time other than for the snapcast provider (and that is not a priority either). But things do change so this will stay here as a reminder.

**Reply by Nocturna22** [reply]
> Both of those things are on the roadmap for the future. In regards to the original post that is not something beling considered at this time other than for the snapcast provider (and that is not a priority either). But things do change so this will stay here as a reminder.

Hi, I'm here to kindly remind you of this feature :)

By the way: i cant find a way to implement a Radio URL in Beta anymore.. Will it be added  again?
